rfc/0054-sdk3-management-apis.md

```diff
@@ -1190,8 +1204,13 @@ void CreateDataset(string datasetName, string bucketName, [options])
 
   * `DataverseName` (`string`) - The name of the dataverse to use, default to none. If set then will be used as `CREATE DATASET
     dataverseName.datasetName`. If not set then will be `CREATE DATASET datasetName`.
+    * This name may contain one or more `/`, the `/` must be split on and retokenized before sending.
+    * Retokenizing follows the following rule: `bucket.name/scope.name` becomes `` `bucket.name` ``.`` `scope.name` ``.
+    * Note that there can be multiple instance of `/` within a name and all must be retokenized.
 
   * `Timeout` or `timeoutMillis` (`int`/`duration`) - the time allowed for the operation to be terminated. This is controlled by the client.
+  
+  * `LinkName` (`string`) - The name of the link to use, default to none. If set then is applied in the form of `CREATE DATASET datasetName AT linkName`.
```

Comment:
I think it might be more complicated? The Analytics DDL reference mentions the syntax is different for "remote" links (couchbase) vs "external" links (s3, azure, etc). It looks like for external links we need to say `EXTERNAL DATASET` instead of just `DATASET`. External datasets also have an optional `USING` parameter to specify the path of the dataset relative to the root of the external datastore.

Since a link name along doesn't tell us whether it's a remote couchbase link, i think maybe we need different parameters depending on the link type? Like, `CouchbaseRemoteLinkName` and `ExternalLinkName` & `Using`, where the two link name parameters are mutually exclusive, and `Using` is optional and only allowed if `ExternalLinkName ` is specified.

Reply:
Or perhaps we could discover the link type by doing a lookup by name, since we know the link name and dataverse?

Reply:
Alternatively to that we could add a `CreateExternalDataset` function? The options for that function would have `Using`.